### PR TITLE
Support for Android's D8 compiler

### PR DESCRIPTION
**Pull request by kevinjeon-g**
Adds version 8.1.56 (latest public). The jar file for running D8 is pulled from Google's Maven repository:

https://maven.google.com/web/index.html#com.android.tools:r8

### CODE REVIEW

File: bin/yaml/android-java.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+compilers:
+  android-d8:
+    type: singleFile
+    dir: r8-{name}
+    targets:
+      - name: 8.1.56
+        filename: r8-8.1.56.jar
+        check_file: r8-8.1.56.jar
```
Comment (mattgodbolt) [inline bin/yaml/android-java.yaml:8]: Just as we did with baksmali can we have this depend on a version of java, and then test that the jar executes a "--version" or similar? Hopefully the template/example of baksmali (I made some changes) will help here!

**Review by mattgodbolt — CHANGES_REQUESTED** [review]

File: bin/yaml/android-java.yaml (hunk 1 above)
Comment (kevinjeon-g) [reply bin/yaml/android-java.yaml]: Done, thanks for the changes!

**Review by mattgodbolt — APPROVED** [review]
Thanks!

**Comment by mattgodbolt** [comment]
Your patience has been legendary: thank you!

But, it's installed :tada:

```sh
$ ce_install install android-d8
2023-11-14 02:13:10,517 lib.installation_context INFO     Making uncached requests
2023-11-14 02:13:10,519 lib.ce_install  INFO     Creating multiprocessing pool with 2 workers
Error: Could not find or load main class com.android.tools.r8.D8
Caused by: java.lang.ClassNotFoundException: com.android.tools.r8.D8
Installing compilers/android-d8 8.1.56
2023-11-14 02:13:18,216 lib.installation_context INFO     Fetching https://dl.google.com/android/maven2/com/android/tools/r8/8.1.56/r8-8.1.56.jar (14952059 bytes)
2023-11-14 02:13:19,227 lib.installation_context INFO     100% of https://dl.google.com/android/maven2/com/android/tools/r8/8.1.56/r8-8.1.56.jar
2023-11-14 02:13:19,398 lib.installation_context INFO     Moving from staging (/opt/compiler-explorer/staging/8ff46497-78a6-4c1d-a322-a897840db581/r8-8.1.56) to final destination (/opt/compiler-explorer/r8-8.1.56)
2023-11-14 02:13:19,994 lib.ce_install  INFO     compilers/android-d8 8.1.56 installed OK
1 packages installed OK, 0 skipped, and 0 failed installation

```

**Comment by kevinjeon-g** [comment]
Thanks for all of the support!

**Pull request merged**